### ISSUE TITLE
Deprecation warnings with Ember 3.27

### ISSUE THREAD

**Issue by boris-petrov** [emberjs/ember-classic-decorator#70]
All [these](https://github.com/emberjs/ember-classic-decorator/blob/v2.0.0/vendor/classic-decorator/index.js#L13) usages cause Ember 3.27 to trigger deprecation warnings. It would be nice to have those fixed before Ember 4 as otherwise `ember-classic-decorator` won't be usable with it.

Somewhat related to #63.

**Comment by jherdman** [comment]
This is currently biting ember-metrics too. We're looking into refactor out our usages of `@classic`, but it'd be nice to be able to buy a little time.

**Comment by rwjblue** [comment]
I think you can add something like:

```js
const _Ember = typeof require === 'function' && typeof require.has === 'function' && require.has('ember') ? require('ember') : Ember;
```

Then update those usages to use `_Ember`. I think most/all of those things are still available on the `ember` module's export just fine...

**Comment by lucacorti** [comment]
This is an issue for at least `ember-metrics` on 4.x which depends on `e-c-d`.

**Comment by boris-petrov** [comment]
This has been fixed by #89.
